bezier/src/cubic_bezier.rs

```diff
@@ -183,6 +183,73 @@ impl CubicBezierSegment {
         find_cubic_bezier_inflection_points(self)
     }
 
+    /// Return local y extremum points or None if this curve is monotone.
+    ///
+    /// This returns the advancements along the curve, not the actual y position.
+    pub fn find_y_extremums(&self) -> UpToTwo<f32> {
+        let mut ret = UpToTwo::new();
+        // See www.faculty.idc.ac.il/arik/quality/appendixa.html for an explanation
+        // The derivative of a cubic bezier curve is a curve representing a second degree polynomial function
+        // f(x) = a * x² + b * x + c such as :
+        let a = 3.0 * (self.to.y - 3.0 * self.ctrl2.y + 3.0 * self.ctrl1.y - self.from.y);
+        let b = 6.0 * (self.ctrl2.y - 2.0 * self.ctrl1.y + self.from.y);
+        let c = 3.0 * (self.ctrl1.y - self.from.y);
+
+        println!("a : {} | b : {} | c : {}", a, b, c);
+
+        // If the derivative is a linear function
+        if a == 0.0 {
+            if b == 0.0 {
+                // If the derivative is a constant function
+                if c == 0.0 {
+                    ret.push(0.0);
+                }
+            } else {
+                ret.push(-c / b);
+            }
+            return ret;
+        }
+
+        fn in_range(t: f32) -> bool { t >= 0.0 && t < 1.0 }
+
+        let discriminant = b * b - 4.0 * a * c;
+
+        println!("delta : {} ", discriminant);
+
+
+        // There is no Real solution for the equation
+        if discriminant < 0.0 {
+            return ret;
+        }
+
+        // There is one Real solution for the equation
+        if discriminant == 0.0 {
+            let t = -b / (2.0 * a);
+            if in_range(t) {
+                ret.push(t);
+            }
+            return ret;
+        }
+
+        // There are two Real solutions for the equation
+        let discriminant_sqrt = discriminant.sqrt();
+        println!("delta_sqrt : {} ", discriminant_sqrt);
+
+        let first_extremum = (-b - discriminant_sqrt) / 2.0 * a;
```

Comment:
It's easy to get confused about the precedence of operators. In order to avoid ambiguity, let's put some parentheses around `2.0 * a` here and below.